### PR TITLE
Update Mintmaker staging deployment

### PR DESCRIPTION
**Pull request by qixiang**
1. Add the `pipelines-as-code-secret` to staging env, this secret is required for Mintmaker to communicate with the Konflux PaC GitHub app.
3. Mount CA bundle with `rh-certs` component, it's required for Mintmaker to call Githup APIs.

### CODE REVIEW

**Comment by scoheb** [comment]
I believe this PR should be changed to only deploy to staging first.

Once you have tested staging, you can create an additional PR to promote to production

**Review by scoheb — CHANGES_REQUESTED** [review]
see previous comment

**Comment by qixiang** [comment]
> I believe this PR should be changed to only deploy to staging first.
> 
> Once you have tested staging, you can create an additional PR to promote to production

@scoheb It makes sense, I've updated the PR to remove the production deployment.
To be honest, I'm not sure how the external secret works. Does someone need to prepare the PaC secret in the `appsre-stonesoup-vault` beforehand? And who can help us to verify the deployment?

File: components/mintmaker/staging/pipelines-as-code-secret-path.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+- op: add
+  path: /spec/dataFrom/0/extract/key
+  value: staging/mintmaker/github-app
```
Comment (scoheb) [inline components/mintmaker/staging/pipelines-as-code-secret-path.yaml:4]: you have your own key in the vault?

Comment (qixiang) [reply components/mintmaker/staging/pipelines-as-code-secret-path.yaml]: changed to use the secret from build-service: `staging/build/stone-stage-p01/github-app`.

File: argo-cd-apps/base/member/infra-deployments/mintmaker/mintmaker.yaml [hunk 2]
```diff
@@ -14,7 +14,9 @@ spec:
                 environment: staging
                 clusterDir: ""
```
Comment (scoheb) [inline argo-cd-apps/base/member/infra-deployments/mintmaker/mintmaker.yaml:15]:
```suggestion
                clusterDir: "base"
```

**Review by scoheb — CHANGES_REQUESTED** [review]

**Review by scoheb — APPROVED** [review]

**Comment by hugares** [comment]
PR passed e2e, merging it

**Pull request merged**